docs/developer-guide/api.rst

```diff
@@ -326,3 +326,67 @@ If a CREATE, DROP, or TERMINATE statement returns a command status with state QU
         "status": "SUCCESS",
         "message":"Stream created and running"
       }
+
+
+Terminate A Cluster
+-------------------
+
+If you don't need your KSQL cluster anymore, you can terminate the cluster and clean up the resources using this endpoint. When a server receives a ``TERMINATE CLUSTER`` request at ``/ksql/terminate`` endpoint, it writes a ``TERMINATE CLUSTER`` command into the command topic.
+Note that ``TERMINATE CLUSTER`` request can only be sent via the ``/ksql/terminate`` endpoint and you cannot send it via the CLI.
+When each server in the cluster reads a ``TERMINATE CLUSTER`` command, it takes the following steps:
+
+-Sets the KSQL engine mode to ``NOT ACCEPTING NEW STATEMENTS`` so no new statement will be passed to the engine for execution.
+-Terminates all of persistent and transient queries in the engine along with the required clean up for each query.
+-Deletes all Kafka topics associated with the streams/tables in the KSQL metastore.
+-Deletes the command topic for the cluster.
+
+**Example request**
+
+   .. code:: http
+
+      POST /ksql/terminate HTTP/1.1
+      Accept: application/vnd.ksql.v1+json
+      Content-Type: application/vnd.ksql.v1+json
+
+      {
+        "streamsProperties": {}
```

Comment:
if not supplying any `streamsProperties` I'm assuming this can be dropped, i.e. can we simplify the default case to just 

```
      POST /ksql/terminate HTTP/1.1
      Accept: application/vnd.ksql.v1+json
      Content-Type: application/vnd.ksql.v1+json
```